### ISSUE TITLE
Align cache marker icon sizes 

### ISSUE THREAD

**Issue by fm-sys** [cgeo/cgeo#11752]
@fm-sys the icons in filter are quite small, especially compared to other sections like status. Could they be aligned?

_Originally posted by @ztNFny in https://github.com/cgeo/cgeo/issues/11592#issuecomment-930546304_

Indeed, they should IMHO use the same size as in the quick settings

![Screenshot_20210930_071719_cgeo geocaching_copy_431x404](https://user-images.githubusercontent.com/64581222/135391947-89abc60d-f181-46d3-a590-ff5f429488c7.png)
![Screenshot_20210930_071639_cgeo geocaching_copy_369x345](https://user-images.githubusercontent.com/64581222/135391949-1c7a8ef1-512f-429f-b377-17d281019812.png)

_Originally posted by @fm-sys in https://github.com/cgeo/cgeo/issues/11592#issuecomment-930793314_

**Comment by ztNFny** [comment]
Not sure if it's the same size as quick filter on map, it should be aligned with the icon size in "status" filter.

**Comment by ztNFny** [comment]
if getCacheTypeMarker is used they have the size of the background image (= circle)

**Comment by eddiemuc** [comment]
yes, getCacheTypeMarker is used, and I can see that it uses background_gc to get the drawabe size (it is 28x28dp). Still there's something happening there which does not seem to happen with other drawables...

Digging into this, I modified the StatusFilterView as follows:
* I enlarged the icon space to 50x50dp (so differences in sizing can be observed easier)
* I added a "border" around the icons view (also so differences in sizing can be observed better)
* I replaced an icon (the one for "watchlist" with the TRADITIONAL cache type marker as returned by above method

This is the result:
![image](https://user-images.githubusercontent.com/6909759/135493004-7d53d904-a041-462c-b773-e1a81a687dfd.png)

As you can see, e.g. the "owned" drawable fills its icon view completely, while for the Traditional Cache type drawable there's some space left around the circle.

Any idea why this might happen?

**Comment by ztNFny** [comment]
> * I replaced an icon (the one for "watchlist" with the TRADITIONAL cache type marker as returned by above method

actually while at it you could replace the "Disabled" with a disabled Trad - it's currently using a static icon depicting just that ;-)

Could you push your branch? I can take a look then, easier with the prep you've already done.

**Comment by ztNFny** [comment]
aaah, I know what's happening!

**Comment by ztNFny** [comment]
aaah, I know what's happening! Due to the different shapes the marker background is actually a 30dp image with a 28dp circle. Hexagon is wider than the circle, that's why I had to make the image larger. And for the sake of simplicity in creation I used the same base document for all 3 shapes.

**Comment by eddiemuc** [comment]
> Could you push your branch? I can take a look then, easier with the prep you've already done.

Actually I was hacking this into another branch where I am currently working on the calculator thingy... 
If you still need it I can move it to a dedicated branch -> but you sound as if you already found the problem?

Meanwhile I actually tried a bit to read this "path" language of vector xmls using some googling. It is not so easy, but if I understand it correctly, then the background_gc.xml draws it's circle not until its border but leaves some space. E.g. it has a canvas of 7.9375x7.9375, it draws a circle with middle point in the middle of this canvas (3.9688,3.9688), but the radius of this circle is NOT 3.9688 but only 3.175. So some space is left on all four sides of the circle in the canvas. May this be the reason for the observed effect?

**Comment by eddiemuc** [comment]
![image](https://user-images.githubusercontent.com/6909759/135496042-38b2b753-c431-41a4-8f0a-6648c0fe89c0.png)

**Comment by eddiemuc** [comment]
> > * I replaced an icon (the one for "watchlist" with the TRADITIONAL cache type marker as returned by above method
> 
> actually while at it you could replace the "Disabled" with a disabled Trad - it's currently using a static icon depicting just that ;-)

Ah, I am in the Status Filter, there is no icon for "Disabled" there :-) I think you are referring to the map quick setting view, that's a different place in code.

**Comment by ztNFny** [comment]
unfortunately there is some scaling happening somewhere, so it's not as quick to fix as I was hoping, requiring also some code changes (I have yet to identify)

You could cherry pick https://github.com/ztNFny/cgeo/commit/4300caed6938fa39cccb25bc95ceb77825e4b52d to see whether that fixes the filter display issue - but please don't commit that anywhere yet.

> Ah, I am in the Status Filter, there is no icon for "Disabled" there :-) I think you are referring to the map quick setting view, that's a different place in code.

Right, it's in map

**Comment by eddiemuc** [comment]
I think I know now what happens:
* In the map view and other views you draw a "big circle" (the white background one) which assumingly fills the complete canvas
* On top of that you draw a slightly smaller circle (the one colored with the cache type's color). This circle does not fill its canvas completely of course, because otherwise it would completely fill the white circle underneath it

Now, for the filter view, you use the second (slightly smaller) circle ONLY. Drawing only this circle, it leaves of course a border to each side (which in map marker view would be "filled" with the underlying white circle).

Have currently no good idea how to efficiently fix this though

**Comment by ztNFny** [comment]
Yes, that's what I wanted to say in https://github.com/cgeo/cgeo/issues/11752#issuecomment-931475712
The circle is 28dp on a 30dp canvas - and the reason for that was the process for creating the SVGs. However when I make the circle on a fitting canvas it somehow affects display in the MapMarker which is kinda unexpected, looking into why that's happening right now. Looking at the code it shouldn't be scaled.

**Comment by ztNFny** [comment]
oops, was another scaling issue. I need to finally align all the sizes between SVGs and Vectors...

Fixed by https://github.com/ztNFny/cgeo/commit/9f5be8da9ada1bddff12d78e9a649941970a8ba5
I'll see that I get a few other fixes into that branch before a PR though.

**Comment by fm-sys** [comment]
> 
> 
> oops, was another scaling issue. I need to finally align all the sizes between SVGs and Vectors...
> 
> Fixed by [ztNFny@9f5be8d](https://github.com/ztNFny/cgeo/commit/9f5be8da9ada1bddff12d78e9a649941970a8ba5) I'll see that I get a few other fixes into that branch before a PR though.

I have assigned you (@ztNFny), to make clear you are already working on it...

**Comment by ztNFny** [comment]
I'll open a PR for this and continue the cleanup of drawables in a new issue to at least get this "bugfix" resolved soonish.
